File: py/db/pymongo-example/connections/mongo.py
# imports
import os
from pymongo import MongoClient

# connection string
client = 'mongodb'
host = os.getenv('MONGO_HOST')
port = os.getenv('MONGO_PORT')
user = os.getenv('MONGO_USER')
pw = os.getenv('MONGO_PW')


connection_string = f"{client}://{user}:{pw}@{host}:{port}"

# client
client = MongoClient(connection_string)

# db
db_name = os.getenv('MONGO_DB')
db = client[db_name]

File: js/ts/db/mongodb-notes/connections/mongo.ts
// imports
import mongodb, { MongoClient } from "mongodb";
// connection string
const host = process.env.MONGO_HOST;
const port = process.env.MONGO_PORT;
const user = process.env.MONGO_USER;
const pw = process.env.MONGO_PW;
const dbName = process.env.MONGO_DB;
// url
const url = `mongodb://${user}:${pw}@${host}:${port}/`;
// db
let client: mongodb.MongoClient;
let db: mongodb.Db;
// get connection
export const getConnection = async () => {
  if (!db) {
    client = await MongoClient.connect(url);
    db = client.db(dbName);
  }
  return db;
};
